contentgrid-spring-data-rest/src/main/java/com/contentgrid/spring/data/querydsl/mapping/CollectionFiltersFactory.java

```diff
@@ -33,13 +34,9 @@ class CollectionFiltersFactory {
     private final PathNavigator pathNavigator;
 
     public CollectionFilters createFilters() {
-        return new CollectionFiltersImpl(filters().collect(Collectors.toUnmodifiableMap(
-                CollectionFilter::getFilterName,
-                Function.identity(),
-                (a, b) -> {
-                    throw new IllegalStateException("Duplicate filter name '%s' mapping to paths '%s' and '%s'".formatted(a.getFilterName(), a.getPath(), b.getPath()));
-                }
-        )));
+        LinkedHashMap<String, CollectionFilter<?>> orderedMap = new LinkedHashMap<>();
+        filters().forEachOrdered(filter -> orderedMap.put(filter.getFilterName(), filter));
+        return new CollectionFiltersImpl(orderedMap);
```

Comment:
1. This is no longer an unmodifiable map
2. There is a `Collector` where you can supply the map constructor: https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/stream/Collectors.html#toMap(java.util.function.Function,java.util.function.Function,java.util.function.BinaryOperator,java.util.function.Supplier)
3. This doesn't check for duplicate filter names